src/Psalm/Internal/Analyzer/Statements/Expression/Fetch/ClassConstFetchAnalyzer.php

```diff
@@ -616,39 +594,6 @@ public static function analyze(
                 );
             }
 
-            $declaring_const_id = strtolower($fq_class_name) . '::' . $stmt->name->name;
-
-            if ($codebase->alter_code && !$moved_class) {
-                foreach ($codebase->class_constant_transforms as $original_pattern => $transformation) {
-                    if ($declaring_const_id === $original_pattern) {
-                        [$new_fq_class_name, $new_const_name] = explode('::', $transformation);
-
-                        $file_manipulations = [];
-
-                        if (strtolower($new_fq_class_name) !== strtolower($fq_class_name)) {
-                            $file_manipulations[] = new \Psalm\FileManipulation(
-                                (int) $stmt->class->getAttribute('startFilePos'),
-                                (int) $stmt->class->getAttribute('endFilePos') + 1,
-                                Type::getStringFromFQCLN(
-                                    $new_fq_class_name,
-                                    $statements_analyzer->getNamespace(),
-                                    $statements_analyzer->getAliasedClassesFlipped(),
-                                    null
-                                )
-                            );
-                        }
-
-                        $file_manipulations[] = new \Psalm\FileManipulation(
-                            (int) $stmt->name->getAttribute('startFilePos'),
-                            (int) $stmt->name->getAttribute('endFilePos') + 1,
-                            $new_const_name
-                        );
```

Comment:
I think this piece was useful (unlike the one that tried to rename the class).

Renaming `C::CC` to `C::DD` here
```php
final class C { public const CC = 1; }
$c = C::class;
echo $c::CC;
```
should result in
```php
echo $c::DD;
```
but currently results in
```php
echo $c::CC; // should be DD
```

Reply:
Thanks!